### PR TITLE
Allow to set SecAuditLog

### PR DESCRIPTION
**Pull request by sathieu**

### CODE REVIEW

File: manifests/vhost.pp [hunk 1]
```diff
@@ -397,6 +405,21 @@
     }
   }
 
+  if $modsec_audit_log_file {
```
Comment (bmjen) [inline manifests/vhost.pp]: @sathieu This logic is confusing. You should always check the "manage" boolean first and determine path variables after. Something like:

```
if $mod_audit_log {
  if $modsec_audit_log_file {
    $modsec_audit_log_destination = "${logroot}/${modsec_audit_log_file}"
  } else if $modsec_audit_log_pipe { 
    $modsec_audit_log_destination = $modsec_audit_log_pipe
  } else {
    if $ssl {
       $modsec_audit_log_destination = "${logroot}/${name}_security_ssl.log"
     } else {
       $modsec_audit_log_destination = "${logroot}/${name}_security.log"
     }
} else {
    $modsec_audit_log_destination = undef
}
```

This way you only set variables/paths if `$modsec_audit_log` is set to true.

File: manifests/vhost.pp [hunk 2]
```diff
@@ -243,6 +247,10 @@
     fail("Apache::Vhost[${name}]: 'error_log_file' and 'error_log_pipe' cannot be defined at the same time")
   }
 
+  if $modsec_audit_log_file and $modsec_audit_log_pipe {
```
Comment (bmjen) [inline manifests/vhost.pp]: This validation should be ignored if `$modsec_audit_log` is `false`.

Comment (sathieu) [inline manifests/vhost.pp]: I kept the same logic as access_log and error_log (just above).

File: manifests/vhost.pp (hunk 1 above)
Comment (sathieu) [inline manifests/vhost.pp]: My intention was that you only need to set one attribute to enable logging.

I've changed the code a bit to improve this.

**Pull request merged**